app/uk/gov/hmrc/cdsreimbursementclaimfrontend/models/reimbursement/DutyType.scala

```diff
@@ -0,0 +1,161 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cdsreimbursementclaimfrontend.models.reimbursement
+
+import cats.kernel.Eq
+import julienrf.json.derived
+import play.api.libs.json.OFormat
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.TaxCode
+import uk.gov.hmrc.cdsreimbursementclaimfrontend.models.TaxCode._
+
+import java.util.Locale
+
+sealed trait DutyType extends Product with Serializable
+
+object DutyType {
+
+  case object UkDuty extends DutyType
+  case object EuDuty extends DutyType
+  case object Beer extends DutyType
+  case object Wine extends DutyType
+  case object MadeWine extends DutyType
+  case object LowAlcoholBeverages extends DutyType
+  case object Spirits extends DutyType
+  case object CiderPerry extends DutyType
+  case object HydrocarbonOils extends DutyType
+  case object Biofuels extends DutyType
+  case object MiscellaneousRoadFuels extends DutyType
+  case object Tobacco extends DutyType
+  case object ClimateChangeLevy extends DutyType
+
+  val dutyTypes: List[DutyType] = List(
+    UkDuty,
+    EuDuty,
+    Beer,
+    Wine,
+    MadeWine,
+    LowAlcoholBeverages,
+    Spirits,
+    CiderPerry,
+    HydrocarbonOils,
+    Biofuels,
+    MiscellaneousRoadFuels,
+    Tobacco,
+    ClimateChangeLevy
+  )
+
+  val customDutyTypes: List[DutyType] = List(
+    UkDuty,
+    EuDuty
+  )
+
+  val exciseDutyTypes: List[DutyType] = List(
+    Beer,
+    Wine,
+    MadeWine,
+    LowAlcoholBeverages,
+    Spirits,
+    CiderPerry,
+    HydrocarbonOils,
+    Biofuels,
+    MiscellaneousRoadFuels,
+    Tobacco,
+    ClimateChangeLevy
+  )
+
+  val dutyTypeToTaxCodes: Map[DutyType, List[TaxCode]] = Map(
+    UkDuty                 -> TaxCode.listOfUKTaxCodes,
+    EuDuty                 -> TaxCode.listOfEUTaxCodes,
+    Beer                   -> List(NI407, NI440, NI441, NI442, NI443, NI444, NI445, NI446, NI447, NI473),
+    Wine                   -> List(NI411, NI412, NI413, NI415, NI419), //TODO potential problem with 411 appearing twice!
+    MadeWine               -> List(NI421, NI422, NI423, NI425, NI429),
+    LowAlcoholBeverages    -> List(NI431, NI433, NI435, NI473),
+    Spirits                -> List(NI438, NI451, NI461, NI462, NI463),
+    CiderPerry             -> List(NI431, NI481, NI483, NI485, NI487),
+    HydrocarbonOils        -> List(
+      NI511,
+      NI511,
+      NI520,
+      NI521,
+      NI522,
+      NI540,
+      NI541,
+      NI542,
+      NI551,
+      NI556,
+      NI561,
+      NI570,
+      NI571,
+      NI572
+    ),
+    Biofuels               -> List(NI589, NI595),
+    MiscellaneousRoadFuels -> List(NI591, NI592),
+    Tobacco                -> List(NI611, NI615, NI619, NI623, NI627, NI633),
+    ClimateChangeLevy      -> List(NI99A, NI99B, NI99C, NI99D)
+  )
+
+  def toDutyType(repr: String): Option[DutyType] =
```

Comment:
Why do we need this boilerplate? Why don't define repr in type self?
```
sealed trait DutyType extends Product with Serializable {
   val repr: String
}

case object UkDuty extends DutyType("ukduty")
```
And then filter something like
```
dutyTypes.find(_.repr === repr)
```
And typeToString becomes as simple as
```
def typeToString(dutyType: DutyType): String = dutyType.repr

Reply:
See point 7 in the REFACTOR.md file.

We really need to look at this whole idea of how enumerations/ADTs are used to build out list of checkboxes/radio buttons. At the moment it's not very good for a few reasons we have talked about before. Also when it comes to switching on Welsh translation we need to be able to reference individual checkboxes w/o English references to avoid confusion. That's why I raised point 7 in that file.

Reply:
Why do not to have key then?
```
sealed trait DutyType extends Product with Serializable {
   val key: String
}
```
and use as
```
def typeToString(dutyType: DutyType): String = messages(dutyType.key)
```
You can use Macros to generate list of the items but not match keys in specific order, it must have some binding in frontend model anyways.

Reply:
This will be all addressed when we raised a ticket for point 7 in the REFACTORING.md.